teps/0069-support-retries-for-custom-task-in-a-pipeline.md

```diff
@@ -216,23 +216,36 @@ performance requirements.
 
 ## Design Details
 
+Add an optional `Retries` field of type `int` to `RunSpec`.
 
-## Test Plan
+Add optional `RetriesStatus` field to `RunStatusFields` of type `[]RunStatus`.
 
-<!--
-**Note:** *Not required until targeted at a release.*
+Add a config map entry (default-short-timeout-seconds) to `config-defaults` in
+order to make short timeout configurable.
+
+```yaml
+# default-short-timeout-seconds contains the default number of
+# seconds to wait for custom task to respond, on timeout it is assumed
+# custom task does not support the feature. Currently, it is used to
+# quickly timeout a retry in a custom-task.
+default-short-timeout-seconds: "30"  # 30 seconds
```

Comment:
i'm surprised to see this config option and i have a couple thoughts:

- Is this addressing something already described in the existing proposal, and if so, what? If it isn't, is there a requirement or some other details we can add to explain why this is needed (maybe with an example)
- Is this a custom task specific timeout? if so, maybe we can include "custom task" in the name to make that clear - or maybe this applies to other timeout operations as well?

Reply:
Actually, at the moment it is custom task retry specific. 

But, it is pretty generic too. For example, certain task require a short timeout, for example wait for a response or an update. 

If you think we should make it specific to custom tasks, then happy to do that !

Reply:
hmm I'm hoping we can avoid adding this timeout if at all possible - but also adding timeouts like this has been discussed in the past - for example in https://github.com/tektoncd/pipeline/pull/3385 we added a currently hardcoded value `minimumResourceAge`. if we do end up needing this timeout here, maybe we can merge these into one configurable value - or maybe you could even use `minimumResourceAge` directly (with a new name to reflect its expanded role? something like `expectedControllerResponseTime` or something.

there is some discussion about this in the custom task TEP as well: https://github.com/tektoncd/community/blob/main/teps/0002-custom-tasks.md#initial-update-timeout & https://github.com/tektoncd/community/blob/main/teps/0002-custom-tasks.md#open-questions